libs/island-ui/core/src/lib/CategoryCard/CategoryCard.tsx

```diff
@@ -41,62 +60,155 @@ const colorSchemes = {
   },
 } as const
 
-export const CategoryCard = ({
-  heading,
-  headingAs = 'h3',
-  headingVariant = 'h3',
-  text,
-  href = '/',
-  tags = [],
-  colorScheme = 'blue',
-  truncateHeading = false,
-}: CategoryCardProps) => {
-  const hasTags = Array.isArray(tags) && tags.length > 0
-  const { borderColor, textColor, tagVariant } = colorSchemes[colorScheme]
+export type CategoryCardImage =
+  | {
+      src: string
+      alt: string
+      objectFit?: ObjectFitProperty
+      customImage?: never
+    }
+  | {
+      src?: never
+      alt?: never
+      objectFit?: never
+      customImage?: ReactNode
+    }
+
+const Component = forwardRef<
+  HTMLElement,
+  CategoryCardProps & CategoryCardImage
+>(
+  (
+    {
+      width,
+      stackWidth = STACK_WIDTH,
+      heading,
+      headingAs = 'h3',
+      headingVariant = 'h3',
+      text,
+      href = '/',
+      tags = [],
+      colorScheme = 'blue',
+      truncateHeading = false,
+      src,
+      alt,
+      objectFit = 'contain',
+      customImage,
+      hyphenate = false,
+      autoStack,
+    },
+    ref,
+  ) => {
+    const { borderColor, textColor, tagVariant } = colorSchemes[colorScheme]
 
-  return (
-    <FocusableBox
-      href={href}
-      display="flex"
-      flexDirection="column"
-      paddingY={3}
-      paddingX={4}
-      borderRadius="large"
-      borderColor={borderColor}
-      borderWidth="standard"
-      height="full"
-      background="white"
-      color={colorScheme}
-    >
-      <Text
-        as={headingAs}
-        variant={headingVariant}
-        color={textColor}
-        truncate={truncateHeading}
-        title={heading}
+    const hasTags = Array.isArray(tags) && tags.length > 0
+    const hasImage = (!!src && !!alt) || customImage
```

Comment:
If either `src` or `alt` are invalid then `hasImage` will contain whatever value is in `customImage`.
Is this on purpose?

```js
// node terminal session
> customImage = 'https://example.com/catpic.png'
> src = undefined
> alt = 'image of cat'
> const hasImage = (!!src && !!alt) || customImage
> hasImage
'https://example.com/catpic.png'
```